app/pages/app/app.js

```diff
@@ -343,6 +344,34 @@ export class AppComponent extends React.Component {
     return null;
   }
 
+  renderNavPatientHeader() {
+    const { 
+      patient, 
+      user,
+      permsOfLoggedInUser,
+      context: { trackMetric },
+      clinicFlowActive,
+      selectedClinicId,
+      location: currentPage,
+      query,
+    } = this.props;
+
+    if (!this.isPatientVisibleInNavbar()) return null; // only show on pages with a patient of focus
+    
+    return (
+      <NavPatientHeader 
+        patient={patient} 
+        user={user} 
+        trackMetric={trackMetric} 
+        permsOfLoggedInUser={permsOfLoggedInUser}
+        clinicFlowActive={clinicFlowActive}
+        selectedClinicId={selectedClinicId}
+        currentPage={currentPage}
+        query={query}
+      />
```

Comment:
I have some ideas that would help us reduce prop drilling and help us decouple our components quite a bit, but wouldn't rock the boat too much. This PR is a little big so I won't suggest them here, but I would love to chat about it.

Reply:
I'm sure we're all ears on how to go about addressing it. Feel free to write up a short EDD Google Doc/Confluence, start a Slack thread, and/or just grab some time for a team huddle to get the ball rolling. We've got a long list of things to modernize/improve on and this is as good a place as any to break the static friction and get some momentum on the efforts.